napalm/junos/junos.py

```diff
@@ -862,6 +862,21 @@ def _process_pipe(cmd, txt):
 
     def get_bgp_config(self, group='', neighbor=''):
         """Return BGP configuration."""
+        def _check_nhs(policies):
+            if not isinstance(policies, list):
+                # Make it a list if it is a single policy
+                policies = [policies]
+            for p in policies:
+                policy = junos_views.junos_policy_config_table(self.device)
```

Comment:
That is going to be a massive performance issue: you are executing this request for each policy. I suggest doing the request at the beginning of the method, and just use the result here.

Reply:
@mirceaulinic not sure I understand... I'm executing a request for each policy applied on export on each neighbor (not all policies!), so we know for each neighbor whether there is a policy with an action `next-hop self` applied.

If I take the entire config for the policies (ie: `show policy-options`) I will check each and every single policy, which will be even worst, non ? thanks

Reply:
you can retrieve all the policies and then do the same iteration but getting the policy from the object to retrieved before instead of having to do a request per policy.